### DISCUSSION TITLE
Error C2280 occurred when I add data binding

### DISCUSSION THREAD

**Discussion by sxlllslgh** [microsoft/WindowsAppSDK#1088] [Q&A] [answered]
I am trying to implement a simple data binding with INotifyPropertyChanged to refresh GUI when when value changed.
**MainWindow.idl**:
```
namespace App {
    [default_interface]
    runtimeclass MainWindow : Microsoft.UI.Xaml.Window, Microsoft.UI.Xaml.Data.INotifyPropertyChanged {
        Int32 MyValue;
        MainWindow();
    }
}
```
**MainWindow.xaml.h**:
```cpp
namespace winrt::App::implementation {
    struct MainWindow : MainWindowT<MainWindow>, Microsoft::UI::Xaml::Data::INotifyPropertyChanged {
        MainWindow();
    public:
        int32_t MyValue();
        void MyValue(int32_t newValue);
        winrt::event_token PropertyChanged(Microsoft::UI::Xaml::Data::PropertyChangedEventHandler const& handler);
        void PropertyChanged(winrt::event_token const& token);
    private:
        int32_t myValue;
        winrt::event<Microsoft::UI::Xaml::Data::PropertyChangedEventHandler> propertyChanged;
    }
}
```
**MainWindow.xaml.cpp**:
```cpp
void MainWindow::MyValue(int32_t newValue) {
    if (myValue != newValue) {
        myValue = newValue;
        propertyChanged(*this, Microsoft::UI::Xaml::Data::PropertyChangedEventArgs{ L"MyValue" });
    }
}
winrt::event_token MainWindow::PropertyChanged(Microsoft::UI::Xaml::Data::PropertyChangedEventHandler const& handler) {
    return propertyChanged.add(handler);
}
void MainWindow::PropertyChanged(winrt::event_token const& token) {
    propertyChanged.remove(token);
}
```
**MainWindow.xaml**:
```xml
<TextBlock Text="{x:Bind MyValue, Mode=OneWay}" />
```
But I got errors like:
```
1>XamlTypeInfo.Impl.g.cpp
1>D:\App\App\Generated Files\winrt\base.h(7761,5): error C2280: "void *winrt::Windows::Foundation::IUnknown::operator new(size_t)": attempting to reference a deleted function (Compiling source file MainWindow.xaml.cpp)
1>D:\App\App\Generated Files\winrt\base.h(2028): message : see declaration of "winrt::Windows::Foundation::IUnknown::operator new" (Compiling source file MainWindow.xaml.cpp)
```
How to solve it?

**Comment by shelllet** [comment]
winrt::event<Microsoft::UI::Xaml::Data::PropertyChangedEventHandler> propertyChanged = {nullptr};

**Reply by sxlllslgh** [reply]
Thanks but it not work.

**Comment by marb2000** [comment, marked as answer]
@RealTommyKlein can you help here, please?  Here you can find a repro:
https://github.com/marb2000/BasicBindingCpp

**Reply by RealTommyKlein** [reply]
This is an error in the header file, the `winrt::BasicBindingCpp::implementation::MainWindow` declaration should not include the `INotifyPropertyChanged` interface as that's handled by C++/WinRT.  The `winrt::BasicBindingCpp::implementation::MainWindow` declaration in `MainWindow.xaml.h` should instead look like this:
`    struct MainWindow : MainWindowT<MainWindow>`

That correct declaration is also visible in the `Generated Files/sources/MainWindow.h` file after a build, even if there's a build error (as long as the `.idl` files themselves are valid).

The app also needs an implementation of the `MyValue` property getter, or the app will also get a linker error.

**Reply by sxlllslgh** [reply]
Thanks very much! It works!

**Comment by marb2000** [comment]
Thanks Tommy! Doing these two things fix the issues. @sxlllslgh I have updated the repro so you can see it working: https://github.com/marb2000/BasicBindingCpp
